node-discovery/src/Ergvein/Node/Parsing.hs

```diff
@@ -0,0 +1,93 @@
+module Ergvein.Node.Parsing
+  ( NamedSockAddr(..)
+  , parseSockAddr
+  , parseSockAddrs
+  , parseHostPort
+  , ipAddressParser
+  ) where
+
+import Data.Text (Text)
+import Network.Socket (SockAddr)
+import Control.Monad
+import Control.Applicative
+import Control.Monad.IO.Class
+import Data.Either
+import Data.IP
+import Data.Maybe
+import Network.DNS.Lookup
+import Network.DNS.Resolver
+import Network.DNS.Types
+import Network.Socket
+import Text.Read (readMaybe)
+import qualified Data.List.Safe as LS
+import qualified Data.Text as T
+import qualified Data.ByteString.Char8 as B8
+import Data.Attoparsec.Text
+import Data.Word
+
+data NamedSockAddr = NamedSockAddr {
+  namedAddrName :: Text
+, namedAddrSock :: SockAddr
+} deriving (Eq, Ord)
+
+parseSockAddrs :: (MonadIO m) => ResolvSeed -> PortNumber -> [Text] -> m [NamedSockAddr]
```

Comment:
`parse` wasn't good idea for `IO` function. I suggest to rename them to `resolveSockAddrs`